src/Dialect/Mlir/DialectBuilder.hpp

```diff
@@ -336,6 +345,21 @@ struct LLVMBuilder final : DialectBuilder {
   mlir::FlatSymbolRefAttr getOrInsertSymbolRef(mlir::ModuleOp module,
       llvm::StringRef symName, mlir::Type resultType,
       llvm::ArrayRef<mlir::Type> operandTypes, bool isVarArg = false) const;
+
+  /// Generate code that looks like "if then with optional else" at LLVM.
+  /// The following prototype code will be generated:
+  /// ```
+  /// llvm.condBr cond, ^thenBlock, ^elseBlock
+  /// ^thenBlock:
+  ///   thenBody
+  /// ^elseBlock:
+  ///   elseBody
+  /// ^mainBlock
+  ///   ...
+  /// ```
+  void ifThenElse(mlir::function_ref<mlir::Value(LLVMBuilder &createLLVM)> cond,
+      mlir::function_ref<void(LLVMBuilder &createLLVM)> thenFn,
```

Comment:
Maybe you could create a special name for this type? for example `using` like here `  using ApiRegistry = std::map<RuntimeAPI::API, RuntimeAPI>;`

Reply:
Yes, I created these aliases inside LLVMBuilder:
```C
using voidFuncRef = mlir::function_ref<void(LLVMBuilder &createLLVM)>;
using valueFuncRef = mlir::function_ref<mlir::Value(LLVMBuilder &createLLVM)>;
```
Please let me know if you prefer another name. Thank!